wandb/sdk/integration_utils/auto_logging.py

```diff
@@ -86,21 +88,42 @@ def patch(self, run: "wandb.sdk.wandb_run.Run") -> None:
             original = functools.reduce(getattr, symbol_parts, self.set_api)
 
             def method_factory(original_method: Any):
-                @functools.wraps(original_method)
-                def method(*args, **kwargs):
+                async def async_method(*args, **kwargs):
+                    future = asyncio.Future()
+
+                    async def callback(coro):
+                        try:
+                            result = await coro
+                            loggable_dict = self.resolver(
+                                args, kwargs, result, timer.start_time, timer.elapsed
+                            )
+                            if loggable_dict is not None:
+                                run.log(loggable_dict)
+                            future.set_result(result)
+                        except Exception as e:
+                            logger.warning(e)
+                            future.set_exception(e)
+
+                    with Timer() as timer:
+                        coro = original_method(*args, **kwargs)
+                        asyncio.ensure_future(callback(coro))
+
+                    return await future  # future.result()
+
+                def sync_method(*args, **kwargs):
                     with Timer() as timer:
                         result = original_method(*args, **kwargs)
-                    try:
                         loggable_dict = self.resolver(
                             args, kwargs, result, timer.start_time, timer.elapsed
                         )
                         if loggable_dict is not None:
                             run.log(loggable_dict)
-                    except Exception as e:
-                        logger.warning(e)
-                    return result
+                        return result
```

Comment:
Why did you remove the try/except logic here?

Reply:
Oops! I overlooked that. Thanks for pointing it out.  Fixed it.

Reply:
while you're at it... could you please check what happens if a request fails, e.g. bc of a timeout or something like that? I think we weren't properly handling that, gotta fix... (i'd say in a separate pr:)